test/gtest/cfg.cpp

```diff
@@ -579,3 +580,84 @@ TEST_F(CFGTest, ReachingDefinitionsLoop) {
 
   EXPECT_EQ(expectedResult, getSetses);
 }
+
+TEST_F(CFGTest, StackLatticeFunctioning) {
+  FiniteIntPowersetLattice contentLattice(4);
+  StackLattice<FiniteIntPowersetLattice> stackLattice(contentLattice);
+
+  StackLattice<FiniteIntPowersetLattice>::Element firstStack =
+    stackLattice.getBottom();
+  StackLattice<FiniteIntPowersetLattice>::Element secondStack =
+    stackLattice.getBottom();
+
+  for (size_t i = 0; i < 4; i++) {
+    FiniteIntPowersetLattice::Element temp = contentLattice.getBottom();
+    for (size_t j = 0; j <= i; j++) {
+      temp.set(j, true);
+    }
+    firstStack.push(std::move(temp));
+  }
+
+  for (uint32_t i = 0; i < 4; i++) {
+    FiniteIntPowersetLattice::Element temp = contentLattice.getBottom();
+    for (size_t j = 0; j <= i; j++) {
+      temp.set(j, true);
+    }
+    secondStack.push(std::move(temp));
+  }
+
+  EXPECT_EQ(stackLattice.compare(firstStack, secondStack),
+            LatticeComparison::EQUAL);
+
+  std::stringstream ss;
+  secondStack.pop().print(ss);
+  EXPECT_EQ(ss.str(), "1111");
```

Comment:
Would it be easy to compare equality directly without converting to a string first? If so, that would simplify the test and make it test the real API more directly.

Reply:
I could do this. But one problem is, when comparing the `StackLattice`, I don't know how useful that would be for everything except the result of the LUB, since we are testing the correctness of the API of the lattice, but at the same time our "expected result" is also constructed using the exact same API. Of course, I could ditch the above test and focus on testing LUBs more.